terraform-aws-github-runner/modules/runners/lambdas/runners/src/scale-runners/runners.ts

```diff
@@ -702,12 +757,24 @@ export async function createRunner(runnerParameters: RunnerInputParameters, metr
         Value: runnerParameters.repoName,
       });
     }
+
     if (runnerParameters.orgName !== undefined) {
       tags.push({
         Key: 'Org',
         Value: runnerParameters.orgName,
       });
     }
+
+    /* istanbul ignore next */
+    if (Config.Instance.runnersExtraLabels) {
```

Comment:
FYI

runner labels is equal to the concatenation of the values from:
`Config.Instance.runnersExtraLabels` + `RunnerTypeOptional.labels` + `RunnerType.runnerTypeName`.

`RunnerInputParameters.runnerConfig` needs all this information to build this. So we need to add here the proper labels.

It is NOT necessary to add `Config.Instance.runnersExtraLabels` to instance tags. There is no point on doing so, it is global and applied to all runners, so it is known already. `runnerTypeName` needs to be stored independently, so, it also could not be stored in this tag.

This tag only makes sense to store ` RunnerTypeOptional.labels`.

Please also be aware that this field is optional, so need to handle properly empty arrays and undefined,.

Reply:
I was thinking,

And we can simplify things even more.

Remember: with `runnerTypeName` we can use `gh-runners.ts` -> `getRunnerTypes`. So all the information stored in `RunnerInputParameters.runnerType` might not be necessary to be stored.

This can simplify a lot all the changes.